src/Aspire.Hosting/ApplicationModel/ExecutableResource.cs

```diff
@@ -24,15 +24,29 @@ public class ExecutableResource(string name, string command, string workingDirec
     : Resource(name), IResourceWithEnvironment, IResourceWithArgs, IResourceWithEndpoints, IResourceWithWaitSupport,
     IComputeResource
 {
+    /// <inheritdoc/>
+    public override ResourceAnnotationCollection Annotations { get; } = [
```

Comment:
I would do this in the constructor similar given how easy it is to replace this.